adminapi/src/main/java/io/minio/admin/MinioAdminClient.java

```diff
@@ -613,6 +620,241 @@ public Message getServerInfo() throws IOException, NoSuchAlgorithmException, Inv
+                creds.secretKey(), OBJECT_MAPPER.writeValueAsBytes(addServiceAccountReq)))) {
+      return addServiceAccountReq;
+    }
+  }
+
+  /**
+   * Edit an existing service account.
+   *
+   * <pre>Example:{@code
+   * // Assume policyJson contains below JSON string;
+   * // {
+   * //     "Statement": [
+   * //         {
+   * //             "Action": "s3:GetObject",
+   * //             "Effect": "Allow",
+   * //             "Principal": "*",
+   * //             "Resource": "arn:aws:s3:::my-bucketname/myobject*"
+   * //         }
+   * //     ],
+   * //     "Version": "2012-10-17"
+   * // }
+   * //
+   * }</pre>
+   *
+   * @param accessKey Access key.
+   * @param newStatus New service account status.
+   * @param newSecretKey New secret key.
+   * @param newPolicy New policy as JSON string .
+   * @param newName New service account name.
+   * @param newDescription New description.
+   * @param newExpiration New expiry time , Example : 2023-12-02T15:04:05Z.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws IOException thrown to indicate I/O error on MinIO REST operation.
+   * @throws InvalidCipherTextException thrown to indicate data cannot be encrypted/decrypted.
+   */
+  public void updateServiceAccount(
+      @Nullable String newName,
+      @Nullable String newSecretKey,
+      @Nullable String accessKey,
+      @Nullable UpdateServiceAccountReq.Status newStatus,
+      @Nullable String newPolicy,
+      @Nullable String newDescription,
+      @Nullable String newExpiration)
+      throws NoSuchAlgorithmException, InvalidKeyException, IOException,
+          InvalidCipherTextException {
+    if (newDescription != null && newDescription.length() > 256) {
+      throw new IllegalArgumentException("description must be at most 256 bytes long");
+    }
+    if (accessKey == null || accessKey.isEmpty()) {
+      throw new IllegalArgumentException("access key must be provided");
+    }
+    byte[] policyBytes = null;
+    if (newPolicy != null && !newPolicy.isEmpty()) {
+      policyBytes = newPolicy.getBytes(StandardCharsets.UTF_8);
+    }
+
+    Credentials creds = getCredentials();
+    UpdateServiceAccountReq updateServiceAccountReq =
+        new UpdateServiceAccountReq(
+            newSecretKey, policyBytes, newStatus, newName, newDescription, newExpiration);
+    try (Response response =
+        execute(
+            Method.POST,
+            Command.UPDATE_SERVICE_ACCOUNT,
+            ImmutableMultimap.of("accessKey", accessKey),
+            Crypto.encrypt(
+                creds.secretKey(), OBJECT_MAPPER.writeValueAsBytes(updateServiceAccountReq)))) {}
+  }
+
+  /**
+   * Creates randomly generated credentials of maximum.
+   *
+   * @param length credentials length
+   * @return credential
+   */
+  private String generateCredentials(int length) {
```

Comment:
I don't think this is needed. You could error out if access/secret key is not passed to methods.